--- ../src-base/minecraft/net/minecraft/world/chunk/storage/AnvilSaveConverter.java
+++ ../src-work/minecraft/net/minecraft/world/chunk/storage/AnvilSaveConverter.java
@@ -1,5 +1,6 @@
 package net.minecraft.world.chunk.storage;
 
+import catserver.server.utils.LanguageUtils;
 import com.google.common.collect.Lists;
 import java.io.DataInputStream;
 import java.io.DataOutputStream;
@@ -225,7 +226,12 @@
                         }
                         else
                         {
-                            NBTTagCompound nbttagcompound = CompressedStreamTools.read(datainputstream);
+                            NBTTagCompound nbttagcompound;
+                            try {
+                                nbttagcompound = CompressedStreamTools.read(datainputstream);
+                            }catch (Exception e) {
+                                throw new RuntimeException(LanguageUtils.I18nToString("world.region_corrupt") + regionfile.getFileName().getCanonicalPath());
+                            }
                             datainputstream.close();
                             NBTTagCompound nbttagcompound1 = nbttagcompound.getCompoundTag("Level");
                             ChunkLoader.AnvilConverterData chunkloader$anvilconverterdata = ChunkLoader.load(nbttagcompound1);
